tests/Integration/UI/SettingsPageTest.php

```diff
@@ -43,6 +43,274 @@ public function set_up(): void {
 		self::$settings_page = new Settings_Page( self::$parsely );
 	}
 
+	/**
+	 * Verifies that valid Site ID values are retained when validated.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers \Parsely\UI\Settings_Page::validate_basic_section
+	 * @uses \Parsely\Parsely::__construct
+	 * @uses \Parsely\Parsely::api_secret_is_set
+	 * @uses \Parsely\Parsely::are_credentials_managed
+	 * @uses \Parsely\Parsely::get_api_secret
+	 * @uses \Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::inject_managed_credentials
+	 * @uses \Parsely\UI\Settings_Page::__construct
+	 * @uses \Parsely\UI\Settings_Page::get_logo_default
+	 * @uses \Parsely\UI\Settings_Page::get_obfuscated_value
+	 * @uses \Parsely\UI\Settings_Page::get_unobfuscated_value
+	 * @uses \Parsely\UI\Settings_Page::sanitize_site_id
+	 * @uses \Parsely\UI\Settings_Page::validate_advanced_section
+	 * @uses \Parsely\UI\Settings_Page::validate_options
+	 * @uses \Parsely\UI\Settings_Page::validate_options_post_type_tracking
+	 * @uses \Parsely\UI\Settings_Page::validate_recrawl_section
+	 * @uses \Parsely\UI\Settings_Page::validate_site_id
+	 *
+	 * @group settings-page
+	 * @group settings-page-validation
+	 */
+	public function test_valid_site_id_values_are_retained_when_validated(): void {
+		$options = self::$parsely->get_options();
+
+		$options['apikey'] = 'mydomain.com';
+		$expected          = $options;
+
+		$actual = self::$settings_page->validate_options( $options );
+		self::assertSame( $expected, $actual );
+	}
+
+	/**
+	 * Verifies that invalid Site ID values are emptied when validated.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers \Parsely\UI\Settings_Page::validate_basic_section
+	 * @uses \Parsely\Parsely::__construct
+	 * @uses \Parsely\Parsely::api_secret_is_set
+	 * @uses \Parsely\Parsely::are_credentials_managed
+	 * @uses \Parsely\Parsely::get_api_secret
+	 * @uses \Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::inject_managed_credentials
+	 * @uses \Parsely\UI\Settings_Page::__construct
+	 * @uses \Parsely\UI\Settings_Page::get_logo_default
+	 * @uses \Parsely\UI\Settings_Page::get_obfuscated_value
+	 * @uses \Parsely\UI\Settings_Page::get_unobfuscated_value
+	 * @uses \Parsely\UI\Settings_Page::sanitize_site_id
+	 * @uses \Parsely\UI\Settings_Page::validate_advanced_section
+	 * @uses \Parsely\UI\Settings_Page::validate_options
+	 * @uses \Parsely\UI\Settings_Page::validate_options_post_type_tracking
+	 * @uses \Parsely\UI\Settings_Page::validate_recrawl_section
+	 * @uses \Parsely\UI\Settings_Page::validate_site_id
+	 *
+	 * @group settings-page
+	 * @group settings-page-validation
+	 */
+	public function test_invalid_site_id_values_are_emptied_when_validated(): void {
+		$expected = self::$parsely->get_options();
+		$options  = self::$parsely->get_options();
+
+		$options['apikey'] = 'invalid';
+
+		$actual = self::$settings_page->validate_options( $options );
+		self::assertSame( $expected, $actual );
```

Comment:
To me the function name sounds confusing because there is a word `emptied` here but we are not asserting `''` in test ..... if we follow above suggestion then it will also make things clear here.

Same feedback for other asserts.